server/chat.ts

```diff
@@ -1803,29 +1806,84 @@ export const Chat = new class {
 	 * @param user - the user that sent the message
 	 * @param connection - the connection the user sent the message from
 	 */
-	parse(message: string, room: Room | null | undefined, user: User, connection: Connection) {
+	async parse(message: string, room: Room | null | undefined, user: User, connection: Connection) {
+		Monitor.activeIp = connection.ip;
 		Chat.loadPlugins();
 
 		const initialRoomlogLength = room?.log.getLineCount();
 		const context = new CommandContext({message, room, user, connection});
 		const start = Date.now();
-		const result = context.parse();
-		if (typeof result?.then === 'function') {
-			void result.then(() => {
-				this.logSlowMessage(start, context);
-			});
-		} else {
-			this.logSlowMessage(start, context);
-		}
+		const result = await context.parse();
+		this.logSlowMessage(start, context);
 		if (room && room.log.getLineCount() !== initialRoomlogLength) {
 			room.messagesSent++;
 			for (const [handler, numMessages] of room.nthMessageHandlers) {
 				if (room.messagesSent % numMessages === 0) handler(room, message);
 			}
 		}
-
+		Monitor.activeIp = null;
 		return result;
 	}
+
+	async queue(message: string, room: Room | null, connection: Connection) {
+		const user = connection.user;
+		const now = Date.now();
+		const noThrottle = user.hasSysopAccess() || Config.nothrottle;
+
+		if (message.startsWith('/cmd userdetails') || message.startsWith('>> ') || noThrottle) {
```

Comment:
Should these commands also be exempt from network throttling? That's how it worked in the old system.